File: 2630-alternating-digit-sum/alternating-digit-sum.py
class Solution:
    def alternateDigitSum(self, n: int) -> int:
        oddIndexSum = 0
        evenIndexSum = 0
        for i in str(n)[0 :: 2]:
            evenIndexSum += int(i)
        for i in str(n)[1 :: 2]:
            oddIndexSum += int(i)
        return evenIndexSum - oddIndexSum

        # Other Way of Approach

        # ans = 0
        # for i, c in enumerate(str(n)):
        #     ans += int(c) * (-1) ** i
        # return ans

File: 2630-alternating-digit-sum/alternating-digit-sum.java
class Solution {
    public int alternateDigitSum(int n) {
        String s = String.valueOf(n);
        int ans = 0;
        for (int i = 0; i < s.length(); i++) {
            int digit = s.charAt(i) - '0';
            ans += digit * (int) Math.pow(-1, i);
        }
        return ans;
    }
}
